src/Parser/DemonHunter/Havoc/Modules/Traits/CriticalChaos.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import CritEffectBonus from 'Parser/Core/Modules/Helpers/CritEffectBonus';
+
+
+const CRITICAL_CHAOS_INCREASE = 0.06;
+
+const CRITICAL_CHAOS_SPELLS = [
+	SPELLS.CHAOS_STRIKE_MH_DAMAGE.id,
+	SPELLS.CHAOS_STRIKE_OH_DAMAGE.id,
+	SPELLS.ANNIHILATION_MH_DAMAGE.id,
+	SPELLS.ANNIHILATION_OH_DAMAGE.id,
+];
+
+const MS_BUFFER = 50;
+/*
+* Critical Chaos (Artifact Trait)
+* Increases critical strike damage dealt by Chaos Strike by 6%.
+*/
+
+class CrticalChaos extends Analyzer {
+	static dependencies = {
+		combatants: Combatants,
+		critEffectBonus: CritEffectBonus,
+	}
+
+	rank = 0;
+	damage = 0;
+  lastCritTimestamp = 0;
+
+	on_initialized() {
+ 		this.rank = this.combatants.selected.traitsBySpellId[SPELLS.CRITICAL_CHAOS.id];
+ 		this.active = this.rank > 0;
+
+ 		if (this.active) {
+      this.critEffectBonus.hook(this.getCritEffectBonus.bind(this));
+    }
+ 	}
+
+ 	getCritEffectBonus(critEffectModifier, event) {
+ 		const spellId = event.ability.guid;
+    if (CRITICAL_CHAOS_SPELLS.includes(spellId)) {
+      critEffectModifier += this.rank * CRITICAL_CHAOS_INCREASE;
+    }
+    return critEffectModifier;
+  }
+
+  getDamageContributon(event) {
+    const raw = (event.amount || 0) + (event.absorbed || 0);
+    const rawNormalPart = raw / this.critEffectBonus.getBonus(event);
+    const rawCritDamage = rawNormalPart * CRITICAL_CHAOS_INCREASE;
+    return rawCritDamage;
+  }
+
+  on_byPlayer_damage(event) {
+  	const spellId = event.ability.guid;
+    //Chaos Cleave Attribution
+    if(this.combatants.selected.hasTalent(SPELLS.CHAOS_CLEAVE_TALENT.id) && 
+      event.timestamp < this.lastCritTimestamp + MS_BUFFER && 
+      spellId === SPELLS.CHAOS_CLEAVE_DAMAGE.id) 
+    {
```

Comment:
This is a bit ugly but I am not sure how to make it look nicer

Reply:
I would increase the buffer (sometimes there is up to 100ms between CS and chaos cleave dmg), or just replace the timestamp with a bool that you change based on the last Chaos Strike being a crit or not. Otherwise it looks fine.

Reply:
I would still need some sort of window if i used a bool right? To account for the cleave damage on AoE? Unless I am thinking about this wrong

Reply:
Shouldn't be necessary.  There should always be a chaos strike damage event before a chaos cleave event you can link it too. Even if the log is bugged and there is chaos cleave events without a chaos strike just before (say eonar range issues), I think it's more correct to just guess based on the most recent chaos strike being a crit or not, rather than just discard it.